core/src/scene/styleContext.cpp

```diff
@@ -83,14 +83,28 @@ StyleContext::~StyleContext() {
     duk_destroy_heap(m_ctx);
 }
 
+// Convert a scalar node to a boolean, double, or string (in that order)
+// and for the first conversion that works, push it to the top of the JS stack.
+void pushYamlScalarAsJsObject(duk_context* ctx, const YAML::Node& node) {
+    bool booleanValue = false;
+    double numberValue = 0.;
+    if (YAML::convert<bool>::decode(node, booleanValue)) {
```

Comment:
I think this function is preferable, since it avoids using exceptions. We could change those yaml helpers to use these functions in the future.